### PR TITLE
Add AbstractView and use it as a base class for Callback

### PR DESCRIPTION
**Pull request by mvorisek**
small break: `View::add()` now checks instanceof AbstractView

### CODE REVIEW

File: src/Callback.php [hunk 1]
```diff
@@ -174,4 +147,12 @@ private function getUrlArguments(string $value): array
     {
         return ['__atk_callback' => $this->urlTrigger, $this->urlTrigger => $value];
     }
+
+    protected function _getStickyArgs($triggerBy): array
+    {
+        // DEV NOTE:
```
Comment (mvorisek) [inline src/Callback.php:153]: will be solved in another/breaking PR later...

File: src/CallbackLater.php [hunk 2]
```diff
@@ -21,7 +21,7 @@ class CallbackLater extends Callback
      *
      * @return mixed|null
      */
-    public function set($callback, $args = [])
+    public function set($callback = null, $args = null)
```
Comment (georgehristov) [inline src/CallbackLater.php]: $fx

**Review by georgehristov — COMMENTED** [review]
I like the idea but:
- what is the breaking change?
- what is the benefit?

File: src/CallbackLater.php (hunk 2 above)
Comment (mvorisek) [reply src/CallbackLater.php]: >     * what is the breaking change?

image


>     * what is the benefit?

strict item tree type validation and later abstract nesting for Callback

Comment (mvorisek) [reply src/CallbackLater.php]: `$fx` renamed

**Review by georgehristov — APPROVED** [review]
Fix coding style and I would say we can merge

Will this solve the problem of repeated reloads when using callback to reload a view?

**Comment by mvorisek** [comment]
> Will this solve the problem of repeated reloads when using callback to reload a view?

Quite a long jorney to it, this PR is for about 20% work needed :)

**Pull request merged**